src/Compilers/CSharp/Portable/Lowering/LambdaRewriter/LambdaCapturedVariable.cs

```diff
@@ -65,6 +65,11 @@ private static string GetCapturedVariableFieldName(Symbol variable, ref int uniq
                 {
                     return GeneratedNames.MakeHoistedLocalFieldName(local.SynthesizedKind, uniqueId++);
                 }
+
+                if (local.SynthesizedKind == SynthesizedLocalKind.InstrumentationPayload)
+                {
+                    return GeneratedNames.MakeSynthesizedInstrumentationPayloadLocalName(uniqueId++);
+                }
```

Comment:
The array of method payloads is rooted in PrivateImplementationDetails. Every instrumented method has a local that holds the method's instrumentation data, which is initialized in the method prologue. It be technically possible to avoid the local and have each instrumentation point refer to the instrumentation cell via the root, but that would require an extra array element reference per statement.

Yes, there is a test for instrumenting a lambda. I discovered that I had to add the code above because that test failed during compilation after introducing the local for the instrumentation data.